python/pyarrow/tests/test_feather.py

```diff
@@ -466,3 +466,9 @@ def test_unsupported(self):
         # non-strings
         df = pd.DataFrame({'a': ['a', 1, 2.0]})
         self._assert_error_on_write(df, ValueError)
+
+    @pytest.mark.slow
+    def test_large_dataframe(self):
+        df = pd.DataFrame(np.random.randint(0, 100, size=(400000000, 1)),
```

Comment:
Any data will do here, so you could do `np.arange` here to make this faster rather than generating so much random data